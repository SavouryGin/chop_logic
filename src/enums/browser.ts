export enum Browser {
  Unknown,
  Opera,
  Chrome,
  Safari,
  Firefox,
}
