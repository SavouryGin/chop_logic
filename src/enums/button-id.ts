export enum ButtonID {
  Cancel,
  Navigation,
  ColorTheme,
  Sounds,
  Settings,
  FullScreen,
  Tools,
  Apply,
  Premise,
  Reiteration,
  Replace,
  Delete,
  ImplicationCreation,
  ImplicationDistribution,
  ContradictionRealization,
  ImplicationElimination,
  Assumption,
  NotIntroduction,
  NotElimination,
  AndIntroduction,
  AndElimination,
  OrIntroduction,
  OrElimination,
  ImpliesIntroduction,
  ImpliesElimination,
  EquivIntroduction,
  EquivElimination,
  Shortcut,
  SavePDF,
  ExportXML,
  ImportXML,
  CopyProof,
  PasteProof,
  CutProof,
  Reset,
}
