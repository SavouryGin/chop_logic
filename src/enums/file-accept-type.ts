export enum FileAcceptType {
  Image = 'image/*',
  Video = 'video/*',
  Audio = 'audio/*',
  XML = 'text/xml',
  PDF = 'application/pdf',
  CSV = '.csv',
  Text = 'text/plain',
  Excel = 'application/vnd.ms-excel, application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
}
