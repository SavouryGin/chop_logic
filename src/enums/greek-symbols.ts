export enum GreekSymbol {
  Phi = '\u03A6',
  Psi = '\u03A8',
}
