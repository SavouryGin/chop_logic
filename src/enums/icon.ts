export enum Icon {
  Default = 'chop-icon__check',
  Home = 'chop-icon__home',
  Menu = 'chop-icon__menu',
  Enlarge = 'chop-icon__enlarge2',
  Shrink = 'chop-icon__shrink2',
  Settings = 'chop-icon__cog',
  Delete = 'chop-icon__bin2',
  Up = 'chop-icon__circle-up',
  Down = 'chop-icon__circle-down',
  Right = 'chop-icon__circle-right',
  Left = 'chop-icon__circle-left',
  Sound = 'chop-icon__volume-medium',
  NoSound = 'chop-icon__volume-mute2',
  LightMode = 'chop-icon__sun',
  DarkMode = 'chop-icon__contrast',
  Cancel = 'chop-icon__close',
  Sidebar = 'chop-icon__magic-wand',
  Telegram = 'chop-icon__telegram',
  Github = 'chop-icon__github',
  Mail = 'chop-icon__mail4',
  Propositions = 'chop-icon__cube',
  Predicates = 'chop-icon__cubes',
  TruthTables = 'chop-icon__delicious',
  Syllogisms = 'chop-icon__dice',
  English = 'chop-icon__english-lang',
  Russian = 'chop-icon__russian-lang',
  Required = 'chop-icon__fire',
  Check = 'chop-icon__checkbox-checked',
  Uncheck = 'chop-icon__checkbox-unchecked',
  SelectUp = 'chop-icon__caret-up',
  SelectDown = 'chop-icon__caret-down',
}
