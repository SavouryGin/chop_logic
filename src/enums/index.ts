export enum Icon {
  Default = 'chop-icon__check',
  Home = 'chop-icon__home',
  Menu = 'chop-icon__menu',
  Enlarge = 'chop-icon__enlarge2',
  Shrink = 'chop-icon__shrink2',
  Settings = 'chop-icon__cog',
  Delete = 'chop-icon__bin2',
  Up = 'chop-icon__circle-up',
  Down = 'chop-icon__circle-down',
  Right = 'chop-icon__circle-right',
  Left = 'chop-icon__circle-left',
  Sound = 'chop-icon__volume-medium',
  NoSound = 'chop-icon__volume-mute2',
  LightMode = 'chop-icon__sun',
  DarkMode = 'chop-icon__contrast',
}

export enum Pages {
  Home,
  NotFound,
  Propositions,
  Predicates,
  TruthTables,
  Syllogisms,
}
