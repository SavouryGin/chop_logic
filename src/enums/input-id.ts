export enum InputID {
  DefaultInput,
  LanguageSelect,
  isDarkModeCheckbox,
  isSoundsCheckbox,
  Premise,
  FirstMetaVariable,
  SecondMetaVariable,
}
