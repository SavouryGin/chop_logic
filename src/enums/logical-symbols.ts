export enum LogicalSymbolHexCode {
  Implication = '\u21D2',
  Conjunction = '\u2227',
  Disjunction = '\u2228',
  Negation = '\xAC',
  Equivalence = '\u21D4',
}

export enum LogicalSymbolRawInput {
  Implication = '=>',
  Conjunction = '&',
  Disjunction = '|',
  Negation = '~',
  Equivalence = '<=>',
}
