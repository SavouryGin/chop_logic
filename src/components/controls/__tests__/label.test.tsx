import Label from 'components/controls/label';
import React from 'react';
import { Icon } from 'enums';
import { render, screen } from '@testing-library/react';

describe('Label component:', () => {
  const testProps = {
    text: 'test-label',
    id: 'test-id',
  };

  it('renders the label component', () => {
    render(<Label {...testProps} />);
    const label = screen.getByTestId('label-test-id');
    expect(label).toBeInTheDocument();
    expect(label).toHaveClass('label');
  });

  it('the label has the passed text', () => {
    render(<Label {...testProps} />);
    const text = screen.getByText(testProps.text);
    expect(text).toBeInTheDocument();
    expect(text).toHaveClass('label__text');
  });

  it('displays the required sign', () => {
    render(<Label {...testProps} isRequired />);
    const requiredSign = screen.getByTitle('required');
    expect(requiredSign).toBeInTheDocument();
    expect(requiredSign).toHaveClass('label__asterisk', Icon.Required);
  });

  it('should match the snapshot', () => {
    const { asFragment } = render(<Label {...testProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
