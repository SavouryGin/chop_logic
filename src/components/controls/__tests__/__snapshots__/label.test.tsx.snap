// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component: should match the snapshot 1`] = `
<DocumentFragment>
  <label
    class="label"
    data-testid="label-test-id"
    for="test-id"
  >
    <span
      class="label__text"
    >
      test-label
    </span>
  </label>
</DocumentFragment>
`;
